pkg/controllers/raycluster_webhook.go

```diff
@@ -77,6 +81,30 @@ func (w *rayClusterWebhook) Default(ctx context.Context, obj runtime.Object) err
 		rayCluster.Spec.HeadGroupSpec.Template.Spec.ServiceAccountName = rayCluster.Name + "-oauth-proxy"
 	}
 
+	// add default queue label if not present
```

Comment:
If the RayCluster is owned by a Kueue-managed object (eg an AppWrapepr) then injecting the queue-name label on a RayCluster that does not have one is not correct.   This functionality at least needs to be guarded by configuration on the operator that allows it to be disabled.  To actually be robust it should be looking to see if the RayCluster is not owned by an AppWrapper.

Reply:
I think this might actually warrant a webhook on the `Workload` CRD and checking if `spec.queue` is set. I don't understand why this approach is incorrect however, using a mutating webhook to inject defaults is a normal use case for mutating webhooks and shouldn't cause any issues since the value is changed prior to being applied to etcd meaning it shouldn't cause issues with re-reconciliation loops. Does this have something to do with AppWrapper functionality that I'm missing?

Reply:
My comment comes from Kueue's logic for supporting parent/child relationships for workloads.  It has code in its defaulting webhooks to detect that the object being created is a child of a Kueue managed parent object (encoded by a controller reference) and to not set the default value for `suspend` in a child job.  
https://github.com/kubernetes-sigs/kueue/blob/release-0.8/pkg/controller/jobframework/defaults.go#L23-L34

I'd suggest adding a similar check here to avoid adding a label which is going to have no impact at runtime and might be confusing to people.  If the ray cluster is controlled by an appwrapper, then it's the queue-name that is on the appwrapper that matters.  The queue-name (or lack of one) on the ray cluster does nothing. 

You can't easily call `IsOwnerManagedByKueue(owner)`, but you could either add a check for the appwrapper GVK or just check for whether the ray cluster is controller by anything at all (I think for standalone clusters, this should be no).

Reply:
I think @dgrove-oss isn't questioning the use of the webhook, but raising the point the logic should only apply to the top-level resource managed by Kueue in the ownership graph.

Reply:
I wonder if the functionality that this PR adds for RayCluster could be generalised and proposed upstream?

Reply:
An upstream generalization would be nice.  For AppWrappers, we went with adding an optional defaultQueueName to the AppWrapper operator configuration.  It's less customizable (uses a single default queue name for all namespaces), but it covered the use case we had for MLBatch.  If an AppWrapper doesn't have a queue-name and the operator is configured with a default then the webhook injects it.  But it only works for AppWrappers.

Reply:
Ahh, I see thanks @dgrove-oss